trace_decoder/src/decoding_smt.rs

```diff
@@ -0,0 +1,24 @@
+use thiserror::Error;
+
+use crate::{
+    aliased_crate_types::{MptGenerationInputs, SmtGenerationInputs},
+    processed_block_trace_smt::SmtProcessedBlockTrace,
+    types::OtherBlockData,
+};
+
+/// Stores the result of parsing tries. Returns a [TraceParsingError] upon
+/// failure.
+pub type SmtTraceParsingResult<T> = Result<T, SmtTraceParsingError>;
+
+/// Error from parsing an SMT trie.
+#[derive(Clone, Debug, Error)]
+pub enum SmtTraceParsingError {}
+
+impl SmtProcessedBlockTrace {
+    pub(crate) fn into_proof_gen_ir(
+        self,
+        other_data: OtherBlockData,
+    ) -> SmtTraceParsingResult<Vec<SmtGenerationInputs>> {
+        todo!()
```

Comment:
Is this the upcoming pr?

Reply:
Yeah, the next PR adds a bit more in here. The PR after that finishes it.

Reply:
Just a reminder that both MPT / SMT files would need update following #207 and `feat/cancun` being merged to `develop`